invoices/tests/models_test.py

```diff
@@ -0,0 +1,189 @@
+import datetime
+from django.db import IntegrityError
+from django.test import TestCase
+from invoices.models import Invoice, Item
+
+
+class InvoiceData(object):
+    """ Data for invoice model
+    """
+    date = datetime.datetime.now()
+    invoice_data = {
+        'code':"1234ssdsd",
+        'order_id':"12",
+        'invoice_date':date,
+        'terms':"Due of Receipt",
+        'due_date':date + datetime.timedelta(days=5),
+        'notes':"sample notes",
+        'conditions':"condition 1",
+        'date_created':date,
+        'date_updated':date,
+        'status':"Sent"
+    }
+
+    def create_invoice(self):
+        data = Invoice.objects.create(
+                code = self.invoice_data['code'],
+                order_id = self.invoice_data['order_id'],
+                invoice_date = self.invoice_data['invoice_date'],
+                terms = self.invoice_data['terms'],
+                due_date = self.invoice_data['due_date'],
+                notes = self.invoice_data['notes'],
+                conditions = self.invoice_data['conditions'],
+                date_created = self.invoice_data['date_created'],
+                date_updated = self.invoice_data['date_updated'],
+                status = self.invoice_data['status']
+            )
+        return data
+
+
+class ItemData(InvoiceData):
+    """ Data for item model
+    """
+    item_data = {
+        'details':"Detail sample",
+        'quantity':3,
+        'rate':45
+    }
+    def create_item(self):
+        invoice = self.create_invoice()
+        data = Item.objects.create(
+            invoice = invoice,
+            details = self.item_data['details'],
+            quantity = self.item_data['quantity'],
+            rate = self.item_data['rate']
+        )
+        return data
+
+
+class InvoiceTestCaseSuccess(TestCase, InvoiceData):
+    """ Test case for invoice model
+    """
+    def setUp(self):
+        self.create_invoice()
+        self.invoice = Invoice.objects.get(code=self.invoice_data['code'])
+
+    def test_add_invoice(self):
+        self.assertEqual(self.invoice.code, self.invoice_data['code'])
+
+    def test_edit_invoice(self):
+        new_code = "new_code"
+        invoice = self.invoice
+        invoice.code = new_code
```

Comment:
you shouldn't modify the code since it is the representation of the invoice object